### ISSUE TITLE
System copying shortcuts don't work with mitmweb

### ISSUE THREAD

**Issue by billinghamj** [mitmproxy/mitmproxy#2910]
##### Steps to reproduce the problem:

1. `mitmweb` - web browser opens
2. highlight any text on the page
3. use Cmd+C system shortcut (standard for copying text on macOS)

Result: nothing is copied - clipboard remains as-is.

Expected: highlighted text is copied to clipboard.

##### Any other comments? What have you tried so far?

Right clicking and hitting copy works, but this is still a massive usability issue.

This has been an issue since at least v2.0.2 - but maybe earlier, and is still a problem on the latest version.

I have had this problem for about a year, using multiple different Macs, so I don't think this is something unique to my current setup.

Have searched, but there don't seem to be any issues mentioning this, surprisingly.

##### System information

```
Mitmproxy: 3.0.2 
Python:    3.6.4
OpenSSL:   OpenSSL 1.0.2n  7 Dec 2017
Platform:  Darwin-17.4.0-x86_64-i386-64bit
```

**Comment by mhils** [comment]
I can't reproduce this on Windows, but it may be a OSX-specific issue. Can someone confirm?

**Comment by fristonio** [comment]
I also couldn't reproduce it on either of `Ubuntu 16.04` and `Ubuntu 14.04`.

**Comment by billinghamj** [comment]
Yeah will need someone else on macOS to test.

That being said, I’m really surprised nobody else has mentioned it before... have had this issue for a long time.

Am using Safari on the latest version of macOS High Sierra also.

**Comment by kira0204** [comment]
This is OSX specific issue, I tested on `Ubuntu16.04` and `Windows 10` and it worked fine.
Tested on OSX with `Safari` and `Google Chrome` and it didn't work. The bug is present in OSX

My wildest guess is that the keybindings with the keyboard are not specified in the react app (Ctrl and Command difference), but couldn't look further because of no dev setup on Mac.

System Information
```
Mitmproxy: 3.0.3 
Python:    3.6.4
OpenSSL:   OpenSSL 1.0.2n  7 Dec 2017
Platform:  Darwin-17.3.0-x86_64-i386-64bit
```

**Comment by billinghamj** [comment]
A friend of mine just pointed out this could be something to do with this code: https://github.com/mitmproxy/mitmproxy/blob/721f7cc92ecdeb03b3f507077e7ef29d48d833ca/web/src/js/ducks/ui/keyboard.js

Perhaps this is being ignored on Windows as websites can’t override the behaviour of Ctrl, for example

**Comment by mhils** [comment]
Yes...
https://github.com/mitmproxy/mitmproxy/blob/721f7cc92ecdeb03b3f507077e7ef29d48d833ca/web/src/js/ducks/ui/keyboard.js#L9 looks particularly suspicious - I wonder how we can detect Cmd instead of Ctrl here.

**Comment by billinghamj** [comment]
It looks like `e.metaKey` would work prevent capturing the Windows key and the Mac Command key.

**Comment by mhils** [comment]
It would be fabulous if anyone of you could test this and submit a PR! 😃

**Comment by fristonio** [comment]
I'll give it a try. I have a test tomorrow will do it after that. :)

**Comment by kira0204** [comment]
HI,
sorry to bug you @MatthewShao , did you get to test this as you mentioned on slack? 
I tried `e.metaKey` and tested but no help. Do you think it is something related to React and it's keybindings?

**Comment by MatthewShao** [comment]
Hi @kira0204 , sorry for the late reply. I don't think this is related to React, actually, I believe this is a browser-related problem. I did test this on an iMac, but did not test the how `e.metaKey` works in different browsers. 
I found some js script aims to solve similar issue @[here](https://stackoverflow.com/a/29745905), my suggestion for this issue is trying how the Vanilla JS works in Mac OS, if works fine, please merge the code into `keybord.js`.
Thanks again for bringing this up.

**Comment by xinranlee** [comment]
still not working on MacOS 10.14.3

**Comment by lkjh654** [comment]
The issue still occurs for me:
Mitmproxy: 4.0.4
Python:    3.7.4
OpenSSL:   OpenSSL 1.0.2s  28 May 2019
Platform:  Darwin-18.7.0-x86_64-i386-64bit
Browser: Chrome Version 76.0.3809.132 (Official Build) (64-bit)

**Comment by andrey-yantsen** [comment]
@lkjh654 seems like the fix wasn't released yet. 4.0.4 was released over a year ago, on 31 July 2018, while the fix was merged on 23 January 2019.

**Comment by lkjh654** [comment]
I see. What's the ETA for a new release?

**Comment by uson1x** [comment]
Still happening for me on 5.0.1

macOS Catalina: 10.15.2 (19C57)
Mitmproxy: 5.0.1
Python:    3.7.6
OpenSSL:   OpenSSL 1.1.1d  10 Sep 2019
Platform:  Darwin-19.2.0-x86_64-i386-64bit

I believe that fix hasn't solved this particular problem.

**Comment by mhils** [comment]
Thanks, I'll reopen this then. I can't reproduce it, so someone on macOS needs to take a look. :)

**Comment by moo-im-a-cow** [comment]
@uson1x  try clearing your browser cache

i was just about to comment that i had this problem with 5.0.1,   
looked in the source, app.js line 191, it didnt have metakey   
while i was writing "5.0.1 doesnt have metakey in the source", i thought to clear the chrome cache, and that fixed it

i havent looked at what build tool is used for the javascript, but i know some build tools can add file hash to the request urls to prevent caching,     
so the request to app.js would be "app.js?abcdefghijk", or possibly "app.js?v=5.0.1", each update is seen by the browser cache as a new file

**Comment by uson1x** [comment]
Hi @moo-im-a-cow 

I don't know what has happened, I have not cleared anything manually, maybe it happened automatically. But I retested just now and it suddenly started working!

Thanks for the heads-up. And **huge** thanks to the person who made it work.

Just for the history, I tested using mitmproxy 5.0.1, in Safari 13.0.5 (15608.5.11) running on macOS Catalina 10.15.3 (19D76).

I guess the ticket can be closed then? Unless somebody else still has issues with shortcuts in macOS?
